Comp2023/src/main/java/frc/robot/subsystems/Swerve.java

```diff
@@ -538,6 +542,70 @@ public void drive(Translation2d translation, double rotation, boolean fieldRelat
     }
   }
 
+  public void driveBalance( )
+  {
+    pitch = m_pigeon.getUnadjustedPitch( ).getWPIRotation2d( ).getDegrees( );
+    DataLogManager.log("Balance Off By: " + pitch);
+
+    if (Math.abs(pitch) > 5)
+    {
+      timer.reset( );
+      while (Math.abs(pitch) > 5 || !(timer.hasElapsed(5)))
+      {
+        timer.reset( );
+        balance( );
+        if (timer.hasElapsed(5))
+        {
+          DataLogManager.log("Breaking: Occurs");
+          break;
+        }
+      }
+    }
+    else if (Math.abs(pitch) < 5)
+    {
+      waitandcheck( );
+    }
+  }
+
+  public void balance( )
+  {
+    if (pitch < 0)
+    {
+      drive(new Translation2d(-0.04 * pitch, 0), 0, true, true);
+      pitch = m_pigeon.getUnadjustedPitch( ).getWPIRotation2d( ).getDegrees( );
+    }
+    else if (pitch > 0)
+    {
+      drive(new Translation2d(-0.04 * pitch, 0), 0, true, true);
+      pitch = m_pigeon.getUnadjustedPitch( ).getWPIRotation2d( ).getDegrees( );
+    }
+    else if (-5 < pitch && pitch < 5)
+    {
+      waitandcheck( );
+    }
+  }
+
+  public void WaitUntilCommand(int time)
```

Comment:
Not sure what this is for. It seems to be just creating a new WaitUntilCommand, but that could be done directly by anyone calling this method. (And it starts with a capital letter which is reserved for Class names--not methods)